web/modules/custom/dpl_admin/dpl_admin.install

```diff
@@ -32,3 +41,102 @@ function dpl_admin_requirements(string $phase): array {
 
   return $requirements;
 }
+
+/**
+ * Migrate administrative role to new user.
+ */
+function dpl_admin_update_10000(array &$sandbox) : string {
+  $feedback = [];
+
+  $user_1 = User::load(1);
+  if (!$user_1) {
+    throw new RuntimeException('Unable to load user 1');
+  }
+
+  // Some field values should not been transfered between the users as we
+  // intentionally want them to be unique.
+  $excluded_fields = ['uid' => TRUE, 'uuid' => TRUE, 'roles' => TRUE];
+  $user_1_fields = array_diff_key($user_1->getFields(include_computed: FALSE), $excluded_fields);
+  $user_1_field_values = array_map(function (FieldItemListInterface $field) {
+    return $field->value;
+  }, $user_1_fields);
+
+  $result = \Drupal::database()->select('users_field_data')
+    ->fields('users_field_data', ['pass'])
+    ->condition('uid', $user_1->id())
+    ->execute();
+  if (!$result) {
+    throw new RuntimeException('Unable to retrieve password hash for user 1');
+  }
+  $password_hash = $result->fetchField();
+  $feedback[] = "Retrieved password hash for user 1";
+
+  // Transfer field data.
+  $new_user = (User::create($user_1_field_values))
+    ->addRole('local_administrator');
+
+  // Reset the existing user 1 account.
+  // Use random data for both username and password as this user is only meant
+  // to be used with onetime login links.
+  $random = new Random();
+  $user_1
+    ->set('field_author_name', "Administrator")
+    ->setUsername('admin_' . $random->name())
+    ->setEmail('')
+    ->setPassword($random->string(32));
```

Comment:
you dont reinitialize `$random`, after having used it for username. could this be a problem?

Reply:
I do not think so. There is no state within [the Random class](https://github.com/drupal/drupal/blob/10.1.x/core/lib/Drupal/Component/Utility/Random.php).

It passes calls to [`mt_rand()`](https://www.php.net/manual/en/function.mt-rand.php) which on the other hand states that it does not provide unguessable values. That might be an issue.

@spaceo, do you have input here?

Reply:
I have replaced `Random` with [`DefaultPasswordGenerator`](https://github.com/drupal/drupal/blob/10.1.x/core/lib/Drupal/Core/Password/DefaultPasswordGenerator.php) which is also used by Drupal Core when creating new passwords. 

This uses [`random_int()`](https://www.php.net/manual/en/function.random-int.php) instead of `mt_rand()` which is also suggested by PHP. As such I think this approach is good.